src/mjolnir/admin.cc

```diff
@@ -29,7 +29,14 @@ Geometry::~Geometry() {
 }
 
 bool Geometry::intersects(const PointLL& ll) const {
+#if GEOS_VERSION_MINOR < 12
+  auto* p = GEOSGeom_createPointFromXY_r(context.get(), ll.lng(), ll.lat());
+  bool intersects = GEOSPreparedIntersects_r(context.get(), prepared, p);
+  GEOSGeom_destroy_r(context.get(), p);
+  return intersects;
+#else
   return GEOSPreparedIntersectsXY_r(context.get(), prepared, ll.lng(), ll.lat());
```

Comment:
could simplify by just removing the `ifdef` and this convenience method in newer versions

Reply:
oh i see you purposely added it so people could use it. i mildly dislike the complexity but its not so bad in this case

Reply:
I was on the fence about it, but ultimately I don't know what GEOS does under the hood (is is just syntactic sugar or does it actually not construct a GEOSGeometry?). Also if it's ugly it's likelier we'll get rid of it once we can? :smile:

Reply:
https://github.com/libgeos/geos/blob/06f74885cb1fd6790bd9286ee2cd8433a349b890/capi/geos_ts_c.cpp#L3842-L3848

yes its mainly syntax sugar, except they keep a point around to reuse rather than allocate and deallocate